File: LcWithPatterns/TypeScript/Stack/lc1189.ts
function maxNumberOfBalloons(text: string): number {
   const counts = { b: 0, a: 0, l: 0, o: 0, n: 0 };

    for(const char of text){
        if (char in counts){
            counts[char as keyof typeof counts]++;
        }
    }

    counts.l = Math.floor(counts.l / 2)
    counts.o = Math.floor(counts.o / 2)

    return Math.min(counts.b, counts.a, counts.l, counts.o, counts.n)
};

File: LcWithPatterns/Python/Stack/lc1189.py
def maxNumberOfBalloons(text: str) -> int:
    counts = {'b': 0, 'a': 0, 'l': 0, 'o': 0, 'n': 0}

    for char in text:
        if char in counts:
            counts[char] += 1

    counts['l'] = counts['l'] // 2
    counts['o'] = counts['o'] // 2

    return min(counts['b'], counts['a'], counts['l'], counts['o'], counts['n'])
